cmd/stack_lint_test.go

```diff
@@ -25,669 +23,108 @@ import (
-			lines[i] = restoreLine
-		}
-	}
-
-	output = strings.Join(lines, "\n")
-	err = ioutil.WriteFile(filepath.Join(sandbox.TestDataPath, "test-stack", "image", "Dockerfile-stack"), []byte(output), 0644)
-
-	if err != nil {
-		t.Fatal(err)
-	}
-}
-
-func TestAppsodyMountsMissingInDockerfileStack(t *testing.T) {
+func TestLintWithValidStack(t *testing.T) {
 	sandbox, cleanup := cmdtest.TestSetupWithSandbox(t, true)
 	defer cleanup()
-
-	restoreLine := ""
-	file, err := ioutil.ReadFile(filepath.Join(sandbox.TestDataPath, "test-stack", "image", "Dockerfile-stack"))
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	lines := strings.Split(string(file), "\n")
-
-	for i, line := range lines {
-		if strings.Contains(line, "APPSODY_MOUNTS") {
-			restoreLine = lines[i]
-			lines[i] = "Testing"
-		}
-	}
-	output := strings.Join(lines, "\n")
-	err = ioutil.WriteFile(filepath.Join(sandbox.TestDataPath, "test-stack", "image", "Dockerfile-stack"), []byte(output), 0644)
-
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	args := []string{"stack", "lint"}
-
-	_, err = cmdtest.RunAppsody(sandbox, args...)
-	if !strings.Contains(err.Error(), "LINT TEST FAILED") {
-		t.Error("String \"LINT TEST FAILED\" not found in output")
-	} else {
-		if err == nil {
-			t.Error("Expected error but did not receive one.")
-		}
-	}
-	for i, line := range lines {
-		if strings.Contains(line, "Testing") {
-			lines[i] = restoreLine
-		}
-	}
-
-	output = strings.Join(lines, "\n")
-	err = ioutil.WriteFile(filepath.Join(sandbox.TestDataPath, "test-stack", "image", "Dockerfile-stack"), []byte(output), 0644)
-
-	if err != nil {
+	args := []string{"stack", "lint", filepath.Join(sandbox.TestDataPath, "test-stack")}
+	_, err := cmdtest.RunAppsody(sandbox, args...)
+	if err != nil { //Lint check should pass, if not fail the test
 		t.Fatal(err)
 	}
 }
-
-func TestAppsodyWatchDirPresentAndOnChangeMissingInDockerfileStack(t *testing.T) {
-	sandbox, cleanup := cmdtest.TestSetupWithSandbox(t, true)
-	defer cleanup()
-
-	restoreLine := ""
-	file, err := ioutil.ReadFile(filepath.Join(sandbox.TestDataPath, "test-stack", "image", "Dockerfile-stack"))
-
-	if err != nil {
-		t.Fatal(err)
-	}
-
-	lines := strings.Split(string(file), "\n")
-
-	if strings.Contains(string(file), "APPSODY_WATCH_DIR") {
-		if strings.Contains(string(file), "_ON_CHANGE") {
-			for i, line := range lines {
-				if strings.Contains(line, "_ON_CHANGE") {
-					restoreLine = lines[i]
-					lines[i] = "Testing"
-				}
-			}
-
-			output := strings.Join(lines, "\n")
-			err = ioutil.WriteFile(filepath.Join(sandbox.TestDataPath, "test-stack", "image", "Dockerfile-stack"), []byte(output), 0644)
-
+func TestLinterInvalidValues(t *testing.T) {
+	var linterInvalidValues = []struct {
+		testName     string
+		targetPath   string
+		containsLine string
+		replaceLine  string
+		expectedLog  string
+	}{
+		{"Invalid Run Value", filepath.Join("image", "Dockerfile-stack"), "APPSODY_RUN", "Testing", "Missing APPSODY_RUN"},
```

Comment:
we could store `filepath.Join` in a var?

Reply:
As in use a var rather than `filepath.Join("image", "Dockerfile-stack")` every time? Yeah that'd work, probably should have noticed that myself